frontend/public/components/monitoring.tsx

```diff
@@ -422,6 +425,66 @@ const SilenceTableRow: RowFunction<Silence> = ({ index, key, obj, style }) => {
   );
 };
 
+const alertMessageResources: { [labelName: string]: K8sKind } = {
+  deployment: DeploymentModel,
+  namespace: NamespaceModel,
+  node: NodeModel,
+  pod: PodModel,
+};
+
+const matchCount = (haystack: string, regExpString: string) =>
+  _.size(haystack.match(new RegExp(regExpString, 'g')));
+
+const AlertMessage: React.FC<{ alert: Alert; rule: Rule }> = ({ alert, rule }) => {
+  let alertText = '';
+  let ruleText = '';
+  if (alert.annotations.description) {
+    alertText = alert.annotations.description;
+    ruleText = rule.annotations.description;
+  } else if (alert.annotations.message) {
+    alertText = alert.annotations.message;
+    ruleText = rule.annotations.message;
+  } else {
+    return null;
+  }
+
+  let messageParts: React.ReactNode[] = [alertText];
+
+  // Go through each recognized resource type and replace any resource names that exist in alertText
+  // with a link to the resource's details page
+  _.each(alertMessageResources, (model, label) => {
+    const labelValue = alert.labels[label];
+
+    if (labelValue && !(model.namespaced && _.isEmpty(alert.labels.namespace))) {
+      const tagCount = matchCount(ruleText, `\\{\\{ *\\$labels\\.${label} *\\}\\}`);
+      const resourceNameCount = matchCount(alertText, _.escapeRegExp(labelValue));
+
+      // Don't do the replacement unless the counts match. This avoids overwriting the wrong string
+      // if labelValue happens to appear elsewhere in alertText
+      if (tagCount > 0 && tagCount === resourceNameCount) {
+        const link = (
+          <ResourceLink
+            className="co-resource-item--monitoring-alert"
+            key={model.kind}
+            kind={model.kind}
+            name={labelValue}
+            namespace={model.namespaced ? alert.labels.namespace : undefined}
```

Comment:
Should we avoid trying to render the link at all if `model.namespaced` and `labels.namespace` is not set? Otherwise it will be a broken link.

Reply:
The `!(model.namespaced && _.isEmpty(alert.labels.namespace))` check above is preventing it rendering a link when `model.namespaced` is true, but `alert.labels.namespace` is not set.